### PR TITLE
Respect short label forms

### LINKED ISSUE #1401: Short label form is not respected
#### Software and hardware versions 
Collect v1.10.0 beta 1

#### Problem description
The `short` itext form as described [here](https://opendatakit.github.io/xforms-spec/#languages), shown in some of the examples [here](https://opendatakit.org/help/form-design/examples/) and in the itext value tag types section [here](https://opendatakit.org/help/form-design/guidelines/) is not respected. I believe the intent was for the `short` form to be displayed in the form hierarchy view and for the `long` form to be used in the widgets. @adammichaelwood perhaps you have seen this in some other documentation? The intent is not clearly stated so I'm hypothesizing.

#### Steps to reproduce the problem
Load the form [short-label-test.xml.txt](https://github.com/opendatakit/collect/files/1271382/short-label-test.xml.txt)

#### Expected behavior
I thought the `short` value would be used in the hierarchy view. I may be misunderstanding the intent. @yanokwa perhaps you remember?

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1401 

#### What has been done to verify that this works as intended?
I tested the attached form.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
[ShorLabel.xml.txt](https://github.com/opendatakit/collect/files/1995009/ShorLabel.xml.txt)

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
Hmmm I'm not sure...

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew pmd checkstyle lint findbugs` and confirmed all checks still pass.
- [x] verified that any new UI elements use theme colors so that they work with both light and dark themes.

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java [hunk 1]
```diff
@@ -398,4 +398,14 @@ public void onClick(DialogInterface dialog, int i) {
         alertDialog.setButton(getString(R.string.ok), errorListener);
         alertDialog.show();
     }
+
+    private String getLabel(FormEntryPrompt formEntryPrompt) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java]: You shouldn't need this one -- `FormEntryPrompt` extends `FormEntryCaption`.

**Review by lognaturel — COMMENTED** [review]
Small opportunity to reduce redundancy but otherwise looks good!

File: collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java]: Right, thanks, it's fixed

**Review by lognaturel — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success!**

Verified on Android 4.1, 4.2, 4.4, 5.1, 6.0, 8.1

| long| short|
|---|---|
| screenshot_20180515-114202|screenshot_20180515-114051|

@opendatakit-bot label "behavior verified"
@opendatakit-bot unlabel "needs testing"

**Pull request merged**